<launch>

  <arg name="gui" default="false" doc="(bool) Launch the gazebo GUI"/>
  <arg name="hand" default="true" doc="(bool) Launch the robot with a hand"/>
  <arg name="usetwo" default="true" doc="(bool) Use two WAM arms instead of one"/>
  <arg name="log_level" default="1" doc="(int) orocos log level"/>
  <arg name="debug" default="false" doc="(bool) in debug mode"/>
  <arg name="rviz" default="true" doc="(bool) run rviz"/>
  <arg name="rviz_config" default="$(find lcsr_collab)/config/tf_peg1.rviz" doc="(string) rviz config file for user interface"/>

  <!-- load the WAM arm settings, space nav controller, and user interface -->
  <include file="$(find lcsr_spacenav)/launch/spacenav_rostopics.launch">
    <arg name="usetwo" value="false"/>
  </include>
  <include file="$(find lcsr_collab)/launch/test7_pegs.launch">
    <arg name="gui" value="$(arg gui)" />
    <arg name="hand" value="$(arg hand)" />
    <arg name="log_level" value="$(arg log_level)" />
    <arg name="debug" value="$(arg debug)"/>
    <arg name="rviz" value="$(arg rviz)" />
    <arg name="rviz_config" value="$(arg rviz_config)" />
    <arg name="usetwo" value="$(arg usetwo)" />
  </include>

  <!-- node for moving ring to destination peg -->
  <node name="move_ring_motion"
    pkg="dmp_action"
    type="dmp_motion_server.py" />

  <!-- node for reaching towards ring -->
  <node name="reach_ring_motion"
    pkg="dmp_action"
    type="dmp_motion_server.py" />

  <!-- wait for the ring to be in range with this action -->
  <node name="wait_for_ring"
    pkg="peg_assist_demo"
    type="WaitForRing.py" />

  <!-- startup script to make sure the arms are running the right controllers -->
  <node name="peg_demo_startup"
    pkg="peg_assist_demo"
    type="startup.py" />

  <!-- node for running the plan -->
  <node name="execution_task" pkg="peg_assist_demo" type="task.py">
    <param name="move_ring_file" value="($find peg_assist_demo)/data/move_ring.yaml"/>
    <param name="grab_ring_file" value="($find peg_assist_demo)/data/grab_ring.yaml"/>
  </node>

  <!-- service node for controlling the robot arm, interfaces with MoveIt and orocos -->
  <node name="Arm2Commander" pkg="peg_assist_demo" type="ArmCommander.py">
    <param name="hand_topic" value="/gazebo/w2barrett_manager/hand/cmd"/>
  </node>

  <!-- view execution progress -->
  <node name="smach_viewer"
    pkg="smach_viewer"
    type="smach_viewer.py" />

</launch>
